scenarios.md

```diff
@@ -0,0 +1,190 @@
+
+### Scenario #7 Automated Deploy - Success
+
+A CD solution initiates a deployment to a development environment. The development environment has been previously configured to only accept signed artifacts.
+
+1. A CD solution is triggered to initiate a deployment
+1. The CD solution uses a deployment template to request an orchestrator to run specific images.
+1. The orchestrator receives the request, validates the artifact is signed and the signature is valid *before* it can initiate the deployment.
+1. The signature is considered valid, and the deployment is successful.
+
+**Implications of this requirement:**
+
+- The aspect of being a signed artifact is not germane to the CD solution. The CD solution simply requests a deployment. The enforcement of signature validation is a policy of the execution environment.
+
+### Scenario #8 Automated Deploy - Failure
+
+Same as #7, with the exception that the signature is deemed invalid.
+1. A CD solution is triggered to initiate a deployment.
+2. The CD solution uses a deployment template to request an orchestrator to run a specific image. In this case, the CD solution mistakenly attempted to deploy a dev signed artifact to a production environment.
+3. The production orchestrator receives the request, validates the artifact is signed and recognizes the signature is **NOT** one of the production accepted signatures. As the signature is deemed invalid, the deployment fails.
+
+**Implications of this requirement:**
+- The coordination between a CD and operational environment for how a request and response is managed is not in-scope for artifact signing.
+- An operational environment enforces a policy for which signatures it accepts. Although the artifact signature may be valid, it's NOT the signature required by this operational environment.
+
+### Scenario #9: Multiple Signatures
+
+Customers may require multiple signatures for the following scenarios:
+
+- Validate the artifact is the same as what the vendor provided.
+- Secondarily sign the artifact by the consuming company, attesting to it's validity within their production environment.
+- Signatures represent validations through different dev, staging, production environments.
+
+1. A CI/CD solution builds, signs, pushes and deploys a collection of artifacts to a staging environment.
+1. Once integrations tests are completed, the artifacts are signed with a production signature, copying them to a production registry or production set of repositories.
+1. The Staging and Production container hosts validate the artifacts contain the signatures required for their environment.
+
+**Implications of this requirement:**
+
+- Multiple signatures, including signatures from multiple sources can be associated with a specific artifact.
+
+## Open Discussions
+
+- What is the relationship between a signature, an artifact and a registry?
+- Can signature validation be dependent on an external entity?
```

Comment:
Orthogonally, can signing be done by an external entity, what formats would the signatures be?

Motivation for this is that if there is an existing generic signing service (with HSM integrations, etc.) that has been used to sign packages, can they be used to generate signatures as well?

Reply:
I believe our goal for Notary v2 is to support signing of things. This does imply a specific pattern. However, this seems like more of a spec question. If this could be re-phrased as a scenario, I can pull it back in.